File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mapElements = {}
        
        for i in range(len(nums)):
            currentNumber = nums[i]
            needed = target - currentNumber

            if needed in mapElements:
                return [mapElements[needed], i]

            mapElements[currentNumber] = i

        return []
        

File: Data Structures & Algorithms/move-zeroes/submission-2.cs
public class Solution {
    public void MoveZeroes(int[] nums) {
        int i = 0;

        while (i < nums.Length) {
            while (i < nums.Length && nums[i] != 0) i++;

            int j = i + 1;

            while (j < nums.Length && nums[j] == 0) j++;

            if (j >= nums.Length)
                break;

            (nums[i], nums[j]) = (nums[j], nums[i]);
        }
    }
}